pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -174,6 +177,9 @@ func MakePod(taskRun *v1alpha1.TaskRun, taskSpec v1alpha1.TaskSpec, kubeclient k
 	initContainers := []corev1.Container{*cred}
 	podContainers := []corev1.Container{}
 
+	maxCpuIdx := findMaxResourceRequest(taskSpec.Steps, corev1.ResourceCPU)
+	maxMemIdx := findMaxResourceRequest(taskSpec.Steps, corev1.ResourceMemory)
```

Comment:
Looping through the containers twice is not the most efficient, but this approach seemed like the most readable code. I am happy to make it more efficient if desired.

Reply:
Also, maybe we should also do the same thing for `corev1.ResourceEphemeralStorage`? Thoughts? If so, it would probably makes sense to switch the signature of `findMaxResourceRequest` to take a list of names and return a map from resource names to max indices.

Reply:
@dwnusbaum I do agree on your 2nd comment :+1:

Reply:
Updated to handle ephemeral storage and avoid repeated looping in be66897.